### DISCUSSION TITLE
react-native 0.65.0 - @walletconnect/modal-react-native node_modules package error: Can't find variable React from 'react'

### DISCUSSION THREAD

**Discussion by fandreirebaseit** [WalletConnect/walletconnect-monorepo#3174] [Developer Support] [answered]
Hi,

After installing all the related packages for @walletconnect/modal-react-native package I get a runtime error when trying to use the modal, in the view where the modal is implemented.

my package.json deps versions:
"react-native": "^0.65.0",
"react": "17.0.2",
"typescript": "^3.8.3"
"big-integer": "^1.6.51",
"bigint-polyfill": "^0.1.0",
"@walletconnect/modal-react-native": "^1.0.0-rc.6",
"@react-native-async-storage/async-storage": "^1.15.8",
"react-native-modal": "^13.0.1",
"react-native-svg": "^12.3.0",
"@types/react-native": "^0.65.2",
"@types/react-test-renderer": "^18.0.0",

I have tried to put the latest async-storage, react-native-svg, and typescript 5.0 but they don't really go well with the react-native 0.65.0 and react 17 so I remained to the versions above.
I haven't seen some minimum requirements for react-native version and I am wandering if anyone could help me by telling me if it's because the react-native version and need to update, if the react version doesn't go with  @walletconnect/modal-react-native version or if there are any dissociations between packages versions. 

The error looks like in the photo below, and if I import React from "react" it will go but there are a bunch of errors like that coming after 
from almost all the @walletconnect/modal-react-native node_modules files

<img width="318" alt="Screenshot 2023-07-17 at 13 30 40" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/109270089/1e29920b-3c8f-4752-b9cb-8a1833118fe0">

**Comment by ignaciosantise** [comment, marked as answer]
hey @fandreirebaseit 👋 

Since React 17, there's no need to import React in every jsx file. Can you try updating your babel to this?

babel.config.js

```js
module.exports = {
  presets: [
    ['module:metro-react-native-babel-preset', { useTransformReactJSXExperimental: true }],
  ],
  plugins: [
      [
          '@babel/plugin-transform-react-jsx',
          {
              runtime: 'automatic',
          },
      ],
  ],
};
```
Close your terminal and re-run yarn start --reset-cache

More info [here](https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#whats-a-jsx-transform)

**Reply by fandreirebaseit** [reply]
Thanks very much, yes, that was the problem, now it's solved.
